api/client/import.go

```diff
@@ -16,30 +18,68 @@ import (
 //
 // The URL argument is the address of a tarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz) file. If the URL is '-', then the tar file is read from STDIN.
 //
-// Usage: docker import [OPTIONS] URL [REPOSITORY[:TAG]]
+// Usage: docker import [OPTIONS] [-i URL] [REPOSITORY[:TAG]]
 func (cli *DockerCli) CmdImport(args ...string) error {
-	cmd := cli.Subcmd("import", "URL|- [REPOSITORY[:TAG]]", "Create an empty filesystem image and import the contents of the\ntarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz) into it, then\noptionally tag it.", true)
+	cmd := cli.Subcmd("import", "[REPOSITORY[:TAG]]", "Create an empty filesystem image and import the contents of the\ntarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz) into it, then\noptionally tag it.", true)
+	infile := cmd.String([]string{"i", "-input"}, "", "Read from a tar archive file, instead of STDIN")
 	flChanges := opts.NewListOpts(nil)
 	cmd.Var(&flChanges, []string{"c", "-change"}, "Apply Dockerfile instruction to the created image")
-	cmd.Require(flag.Min, 1)
+	cmd.Require(flag.Min, 0)
 
 	utils.ParseFlags(cmd, args, true)
 
 	var (
-		v          = url.Values{}
-		src        = cmd.Arg(0)
-		repository = cmd.Arg(1)
+		input      io.Reader = cli.in
+		err        error
+		src        string
+		repository string = cmd.Arg(1)
+		v                 = url.Values{}
 	)
+	if *infile != "" {
+
+		repository = cmd.Arg(0)
+
+		if strings.HasPrefix(*infile, "http://") || strings.HasPrefix(*infile, "https://") {
+			src = *infile
+			input = nil
+		} else if strings.HasPrefix(*infile, "file://") {
+			fileUrl := *infile
+			input, err = os.Open(fileUrl[7:])
+			src = "-"
+			if err != nil {
+				return err
+			}
+		} else {
+			input, err = os.Open(*infile)
+			src = "-"
+			if err != nil {
+				return err
+			}
+		}
+
+	} else {
+		repository = cmd.Arg(0)
+		src = "-"
+
+		if cmd.NArg() > 1 || strings.HasPrefix(cmd.Arg(0), "http://") || strings.HasPrefix(cmd.Arg(0), "https://") {
+			fmt.Fprintf(cli.err, "[DEPRECATED] The format 'URL|- [REPOSITORY[:TAG]]' has been deprecated. Please use [-i URL] [REPOSITORY[:TAG]]\n")
```

Comment:
Actually, the more I look at this the more reversed it seems :-)
You're adding `-i` and moving normal http(s) references under that.  It seems to me that if you wanted to push people towards using a URL for everything (which I assume is why you want to use file://) then we shouldn't have a `-i` at all and the normal URL variant should just support `file://`. 

I would actually prefer to NOT have a -i at all and just have the syntax be:
```
docker import file|URL|- [REPOSITORY[:TAG]]
```
And assume that if it doesn't start with `http(s)://` , or its not `-`, then its must be a local file.